### PR TITLE
Fix convection issues when meteorology is Grell Freitas scheme

### PR DESCRIPTION
**Pull request by yuanjianz**
### Name and Institution (Required)

Yuanjian Zhang (WashU), Lizzie Lundgren (Harvard)

### Describe the update

See discussion https://github.com/geoschem/geos-chem/issues/1409

GEOS-IT and GEOS-FP after Jun 2020 switch to Grell Freitas convection from RAS. Some meteorology field definitions are changed. This update accommodates these fields to geos-chem convection.

Reported by @viral211 first:
>There was an error in archiving the convective precipitation fields (PFLCU and PFICU) in our GEOS-IT and GEOS-FP data after the switch to Grell-Freitas (the fields are just set to zero). These fields are used in GEOS-Chem to calculate convective washout and re-evaporation

@yuanjianz:
>Since DQRCU by definition is changed to net precipitation (MERRA-2 definition DQRCU minus layer-wise re-evaporation), do you think we should add layer-wise REEVAP back to match the DQRCU for determining cloud base height? It may lower the cloud base height somehow, given it seems some gas tracer is substantially below observation constraint?

Changes include:
1. Add surface precipitation flux fields as inputs to GCHP.
2. Add meteorology source name to GCHP run directory `geoschem_config.yml`
3. Create new `Input_Opt` logical for whether to reconstruct convective precipitation flux. This switch is set automatically during run-time to true for GEOS-IT, true for GEOS-FP if start date is June 1 2020 or later, and false for all other cases.
4. Use the new `Input_Opt` logical to use an alternative method in convection which bypasses use of PFLCU and PFICU when necessary. This fixes the zero convective precipitation and high cloud base in runs using GEOS-FP (>=01Jun2020) or GEOS-IT.
5. Add a warning during run directory creation about the convection scheme change and the precipitation flux bug (only if selecting GEOS-FP meteorology). This warning stops run directory creation and requires acknowledgement from the user to continue.

Related convection fixes are also applicable to simulations using met-fields generated with RAS. These are submitted via separate PR https://github.com/geoschem/geos-chem/pull/2573. Those updates will impact the GEOS-Chem benchmark.

### Expected changes

Lower cloud base and restored washout in geos-chem convection driven by GEOS-IT and GEOS-FP after Jun 2020. This will not affect the GEOS-Chem benchmark which uses MERRA2.

### Related Github Issue

https://github.com/geoschem/geos-chem/issues/2469

### CODE REVIEW

File: GeosCore/convection_mod.F90 [hunk 1]
```diff
@@ -616,24 +652,29 @@ SUBROUTINE DO_CLOUD_CONVECTION( Input_Opt,  &
...
+    ! PFLCU and PFICU are converted from kg/m2/s to m3/m2/s
+    ! using water and ice densities, respectively.
+    ! m3/m2/s becomes cm3/cm2/s using a factor of 100.
+    IF ( Is_GF_Conv == .FALSE. ) THEN
+       PDOWN(:) = ( ( PFLCU(:) / 1000e+0_fp ) &
+                +   ( PFICU(:) /  917e+0_fp ) ) * 100e+0_fp
+    ELSE
+       PDOWN(NLAY) = 0
+       DO K = NLAY-1, 1, -1
+          PDOWN(K) = PDOWN(K+1) + DQRCU_MET(K+1) &
```
Comment (viral211) [inline GeosCore/convection_mod.F90:667]: Shouldn't this be: PDOWN(K) = PDOWN(K+1) + DQRCU_MET(**K**) * DELP(**K**) * G0_100 * 100e+0_fp

Comment (yuanjianz) [reply GeosCore/convection_mod.F90:667]: I did this to align with the pre-existing configuration:
https://github.com/geoschem/geos-chem/blob/bef56c605e018eecbd91646a51ce82c7cd77f56a/GeosCore/convection_mod.F90#L569-L575
But if we look at where PDOWN is used, it seems to imply that PDOWN(K) to be the flux at the bottom of level K. So I think you are right, and we need to fix both RAS and GF's PDOWN. 

Also, I have a question about:
https://github.com/geoschem/geos-chem/blob/bef56c605e018eecbd91646a51ce82c7cd77f56a/GeosCore/convection_mod.F90#L1101-L1107
Why is essential to have re-evaporation to have wash-out? Couldn't it be washout event without re-evaporation?

Comment (viral211) [reply GeosCore/convection_mod.F90:667]: You are right. Re-evaporation should not be a condition for washout, but I guess there will always be re-evaporation when the precip falls through drier air below the cloud base. Could you remove the re-evaporation condition and see how much difference it makes? 
Let's fix PDOWN in RAS too. Thanks!

Comment (yuanjianz) [reply GeosCore/convection_mod.F90]: Hi Viral, please check the latest commit.

Comment (viral211) [reply GeosCore/convection_mod.F90]: Hi @yuanjianz, this looks great! One thing I could not tell for certain is whether we go through the entire routine even if there is no convection taking place in a column at the time step, or if there is an early exit somewhere if, say, the convective mass flux is zero throughout the column.

**Comment by lizziel** [comment]
I updated the PR description to include my contributions submitted via https://github.com/yuanjianz/geos-chem/pull/1.

File: run/GCClassic/createRunDir.sh [hunk 2]
```diff
@@ -353,6 +353,26 @@ while [ "${valid_met}" -eq 0 ]; do
...
+
+	# Print warning about GEOS-FP and require user to acknowledge it.
+	fp_msg="WARNING: The convection scheme used to generate archived GEOS-FP meteorology \nfiles changed from RAS to Grell-Freidas starting June 1 2020 with impact on \nvertical transport. Discussion and analysis of the impact is available at \ngithub.com/geoschem/geos-chem/issues/1409. In addition, there is a bug in \nconvective precipitation flux following the switch where all values are zero \nin the input files. This bug is addressed by computing fluxes online for runs \nstarting on or after June 1 2020. The fix does not extend to the case of running \nacross the time boundary. Due to these issues we recommend splitting up GEOS-FP \nruns in time such that a single simulation does not span the switch. Configure \none run to end on June 1 2020 and then use its output restart to start another \nrun on June 1. Alternatively consider using MERRA2. If you wish to use a \nGEOS-FP meteorology year different from your simulation year please create a \nGEOS-Chem GitHub issue for assistance to avoid accidentally using zero \nconvective precipitation flux.\n"
```
Comment (yantosca) [inline run/GCClassic/createRunDir.sh:358]: Fix typo: "Grell-Freidas" -> "Grell-Freitas"

File: run/GCHP/createRunDir.sh [hunk 3]
```diff
@@ -317,6 +317,25 @@ while [ "${valid_met}" -eq 0 ]; do
 
        	met="geosfp"
 
+	# Print warning about GEOS-FP and require user to acknowledge it.
+	fp_msg="WARNING: The convection scheme used to generate archived GEOS-FP meteorology \nfiles changed from RAS to Grell-Freidas starting June 1 2020 with impact on \nvertical transport. Discussion and analysis of the impact is available at \ngithub.com/geoschem/geos-chem/issues/1409. In addition, there is a bug in \nconvective precipitation flux following the switch where all values are zero \nin the input files. This bug is addressed by computing fluxes online for runs \nstarting on or after June 1 2020. The fix does not extend to the case of running \nacross the time boundary. Due to these issues we recommend splitting up GEOS-FP \nruns in time such that a single simulation does not span the switch. Configure \none run to end on June 1 2020 and then use its output restart to start another \nrun on June 1. Alternatively consider using MERRA2. If you wish to use a \nGEOS-FP meteorology year different from your simulation year please create a \nGEOS-Chem GitHub issue for assistance to avoid accidentally using zero \nconvective precipitation flux.\n"
```
Comment (yantosca) [inline run/GCHP/createRunDir.sh:321]: Fix typo: "Grell-Freidas" -> "Grell-Freitas"

**Review by yantosca — CHANGES_REQUESTED** [review]
Hi @lizziel, thanks for the PR.  I just caught a couple of typos in the warning messages in the rundir creation scripts.  Otherwise looks good to go!

**Review by yantosca — APPROVED** [review]
Thanks @lizziel and @yuanjianz!  Good to merge.

**Comment by lizziel** [comment]
Great, I'll bring this in along with the GCHPctmEnv fixes PR.

**Pull request merged**